Source/JavaScriptCore/runtime/IntlDurationFormat.cpp

```diff
@@ -0,0 +1,776 @@
+{
+    return Structure::create(vm, globalObject, prototype, TypeInfo(ObjectType, StructureFlags), info());
+}
+
+IntlDurationFormat::IntlDurationFormat(VM& vm, Structure* structure)
+    : Base(vm, structure)
+{
+}
+
+void IntlDurationFormat::finishCreation(VM& vm)
+{
+    Base::finishCreation(vm);
+    ASSERT(inherits(info()));
+}
+
+enum class StyleListKind : uint8_t { LongShortNarrow, LongShortNarrowNumeric, LongShortNarrowNumericTwoDigit  };
+static IntlDurationFormat::UnitData intlDurationUnitOptions(JSGlobalObject* globalObject, JSObject* options, TemporalUnit unit, PropertyName propertyName, PropertyName displayName, IntlDurationFormat::Style baseStyle, StyleListKind styleList, IntlDurationFormat::UnitStyle digitalBase, std::optional<IntlDurationFormat::UnitStyle> prevStyle)
+{
+    // https://tc39.es/proposal-intl-duration-format/#sec-getdurationunitoptions
+
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    std::optional<IntlDurationFormat::UnitStyle> styleValue;
+    switch (styleList) {
+    case StyleListKind::LongShortNarrow: {
+        styleValue = intlOption<std::optional<IntlDurationFormat::UnitStyle>>(globalObject, options, propertyName, { { "long"_s, IntlDurationFormat::UnitStyle::Long }, { "short"_s, IntlDurationFormat::UnitStyle::Short }, { "narrow"_s, IntlDurationFormat::UnitStyle::Narrow } }, "style must be either \"long\", \"short\", or \"narrow\""_s, std::nullopt);
+        RETURN_IF_EXCEPTION(scope, { });
+        break;
+    }
+    case StyleListKind::LongShortNarrowNumeric: {
+        styleValue = intlOption<std::optional<IntlDurationFormat::UnitStyle>>(globalObject, options, propertyName, { { "long"_s, IntlDurationFormat::UnitStyle::Long }, { "short"_s, IntlDurationFormat::UnitStyle::Short }, { "narrow"_s, IntlDurationFormat::UnitStyle::Narrow }, { "numeric"_s, IntlDurationFormat::UnitStyle::Numeric } }, "style must be either \"long\", \"short\", \"narrow\", or \"numeric\""_s, std::nullopt);
+        RETURN_IF_EXCEPTION(scope, { });
+        break;
+    }
+    case StyleListKind::LongShortNarrowNumericTwoDigit: {
+        styleValue = intlOption<std::optional<IntlDurationFormat::UnitStyle>>(globalObject, options, propertyName, { { "long"_s, IntlDurationFormat::UnitStyle::Long }, { "short"_s, IntlDurationFormat::UnitStyle::Short }, { "narrow"_s, IntlDurationFormat::UnitStyle::Narrow }, { "numeric"_s, IntlDurationFormat::UnitStyle::Numeric }, { "2-digit"_s, IntlDurationFormat::UnitStyle::TwoDigit } }, "style must be either \"long\", \"short\", \"narrow\" or \"numeric\", or \"2-digit\""_s, std::nullopt);
+        RETURN_IF_EXCEPTION(scope, { });
+        break;
+    }
+    }
+
+    IntlDurationFormat::Display displayDefault = IntlDurationFormat::Display::Always;
+    IntlDurationFormat::UnitStyle style = styleValue.value_or(IntlDurationFormat::UnitStyle::Long);
+    if (!styleValue) {
```

Comment:
Seems weird to use `value_or` if we're guaranteed not to use the default value.
```suggestion
    IntlDurationFormat::UnitStyle style;
    if (styleValue)
        style = styleValue.value();
    else {
```

Reply:
Changed.